app/controllers/comment_controller.rb

```diff
@@ -30,12 +34,36 @@ def create
           end
         end
       end
-    else
+    rescue CommentError
       flash[:error] = 'The comment could not be saved.'
       render text: 'failure'
     end
   end
 
+  def create_by_token
+    @node = Node.find params[:id]
+    @user = User.find_by_username params[:username]
+    @body = params[:body]
+    @token = request.env['rack.session']['token']
+
+    if @user && @user.token == @token
+      begin
+        @comment = create_comment(@node, @user, @body)
```

Comment:
Maybe add a comment here telling where to find this method? I think for newcomers the Rails organization can be a bit arcane and although it adds a line, this could be really helpful for some.